utils/run_test.sh

```diff
@@ -79,6 +79,32 @@ if [ -d "/mnt/daos" ]; then
         SL_PREFIX=$PWD/${SL_PREFIX/*\/install/install}
     fi
 
+    echo "Running Cmocka tests"
+    if [ -z "$RUN_TEST_VALGRIND" ]; then
```

Comment:
in what situations will this check be false (and the go tests be skipped)? during which CI runs will RUN_TEST_VALGRIND be set?

Reply:
we plan to add a new parallel step to the run_test.sh stage that does valgrind

Reply:
@ashleypittman if you can give this a +1, we'll look at doing that